lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -540,15 +580,157 @@ def test_cache_map_bad_dir_path(self, monkeypatch, tmp_path):
                 tmp_path, "dir_name"
             )
             tb.cache_map(tar_dir)
-
-            file_path = "/dir_name/subdir1/f11.txt"
-            expected_msg = f"The directory path {file_path!r} is an absolute path, we expect relative path to the root directory."
             with pytest.raises(BadDirpath) as exc:
                 tb.get_info(Path(file_path))
             assert str(exc.value) == expected_msg
 
-    def test_cache_map_traverse_get_info_cmap(self, monkeypatch, tmp_path):
-        """Test to build the cache map of the unpacked tarball"""
+    @pytest.mark.parametrize(
+        "file_path, expected_msg",
+        [
+            ("dir_name/subdir1/./f11.txt", "f11.txt"),
+            ("dir_name/subdir1//f11.txt", "f11.txt"),
+        ],
+    )
+    def test_cache_map_irregular_path(
+        self, monkeypatch, tmp_path, file_path, expected_msg
+    ):
+        """Test to check irreguraties in file/directory path"""
+        tar = Path("/mock/dir_name.tar.xz")
+        cache = Path("/mock/.cache")
+
+        with monkeypatch.context() as m:
+            m.setattr(Tarball, "__init__", TestCacheManager.MockTarball.__init__)
+            m.setattr(Controller, "__init__", TestCacheManager.MockController.__init__)
+            tb = Tarball(tar, Controller(Path("/mock/archive"), cache, None))
+            tar_dir = TestCacheManager.MockController.generate_test_result_tree(
+                tmp_path, "dir_name"
+            )
+            tb.cache_map(tar_dir)
+            tb.get_info(Path(file_path))
+            assert (
+                tb.cachemap["dir_name"]["children"]["subdir1"]["children"]["f11.txt"][
+                    "details"
+                ].name
+                == expected_msg
+            )
```

Comment:
The call at line 609 is slightly mysterious -- the test does not check its result...so, all we're doing here is making sure that the call doesn't make the test _crash_...if nothing else, that deserves a code comment.

However, I think it would be better if the test captured its return value and performed an assertion on it.

The assertion at line 610 is really checking the result of the call at line 608, right?  I would be better if that assertion were checked before making the call to `get_info()`.  (Or, if you add an assertion on the value returned by `get_info()`, we could just omit this assertion.)

And, since what this test is trying to check is the cache traversal, it might be better to call `traverse_cmap()` rather than `get_info()`.

Reply:
I believe this `get_info` was not reqired.